WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift

```diff
@@ -227,31 +252,67 @@ class BloggingRemindersFlowSettingsViewController: UIViewController {
-        calendar.locale = Locale.autoupdatingCurrent
-        let firstWeekday = calendar.firstWeekday
-        var symbols = calendar.shortWeekdaySymbols
+        let createButton: (Int) -> CalendarDayToggleButton = { [unowned self] dayIndex in
+            let isSelected: Bool
+            let localizedDayIndex = (dayIndex + calendar.firstWeekday - 1) % calendar.shortWeekdaySymbols.count
+
+            let weekday = BloggingRemindersScheduler.Weekday(rawValue: localizedDayIndex) ?? {
+                // The rawValue above should always fall within 0 ..< 7 and this case
+                // should not be possible.  We're still going to handle it by logging
+                // an error and returning .monday so that the app keeps running.
+                //
+                // Is there any other sensible approach to handle this?
+                //
+                DDLogError("Cannot initialize BloggingRemindersScheduler.Weekday with rawValue: \(localizedDayIndex)")
+                return .monday
+            }()
+
+            isSelected = weekdays.contains(weekday)
+
+            return CalendarDayToggleButton(
+                weekday: weekday,
+                dayName: self.calendar.shortWeekdaySymbols[localizedDayIndex],
+                isSelected: isSelected) { button in
+
+                if button.isSelected {
+                    self.weekdays.append(button.weekday)
+                } else {
+                    self.weekdays.removeAll { weekday in
+                        weekday == button.weekday
+                    }
+                }
+            }
+        }
 
-        // Switch around the order of days so that the correct day is at the beginning
-        let firstWeekdayToEnd = symbols[firstWeekday-1 ..< Calendar.current.shortWeekdaySymbols.count]
-        let beginningToFirstWeekday = symbols[0 ..< firstWeekday - 1]
+        let topRow = 0 ..< 4
+        let bottomRow = 4 ..< calendar.shortWeekdaySymbols.count
 
-        return Array(firstWeekdayToEnd + beginningToFirstWeekday)
-    }()
+        daysTopInnerStackView.addArrangedSubviews(topRow.map({ createButton($0) }))
+        daysBottomInnerStackView.addArrangedSubviews(bottomRow.map({ createButton($0) }))
+    }
 
     // MARK: - Actions
 
     @objc private func notifyMeButtonTapped() {
         tracker.buttonPressed(button: .continue, screen: .dayPicker)
 
+        do {
+            try scheduler.schedule(.weekdays(weekdays))
+        } catch {
+            DDLogError("Error scheduling notifications: \(error)")
+
+            // TODO: Properly handle error situation.
+            //
+            // For now I'm dismissing the flow in this scenario, although I think showing an inline
+            // error message would be best (an error that's informative and non-blocking, that lets
+            // the user decide if to dismiss or continue).
+            //
+            // I've pinged @osullivanchris about this and will update when I get feedback.
+            //
```

Comment:
same comment about the code comments above

Reply:
As replied above... I disagree.  This is exactly what documentation should look like.

What's being documented is not the history of the decision, but the "why" of a very special scenario which may not be clear by just reading the code.  The question is "why are we doing this?", and the documentation answers that question.

By documenting the "why" we make sure whoever reads the code in the future can understand why it was done this way, and what is required to change it or remove it, and can update the comment if needed.

As I learnt a long time ago, never document the "what", always document the "why".

Just to complete the idea, here's an example of poor code documentation (documenting the "what", instead of the "why"):

```swift
// Assign the last index
let lastIndex = 4
```

That said I'll remove the last line of the documentation as that's already resolved.

Reply:
I ended up updating this anyway, but I'll leave this as unresolved because it describes the way in which I document the code.

Reply:
I was not arguing against the why, just the length of the comment. Too much information in a comment can be confusing, nor I think comments should be used as a todo list (the action to ping someone that you just removed, for example).